src/lib/zkapp.ts

```diff
@@ -583,48 +595,59 @@ class SmartContract {
   }: {
     verificationKey?: { data: string; hash: Field | string };
     zkappKey?: PrivateKey;
-  }) {
+  } = {}) {
     verificationKey ??= (this.constructor as any)._verificationKey;
     if (verificationKey !== undefined) {
       let { hash: hash_, data } = verificationKey;
       let hash = typeof hash_ === 'string' ? Field(hash_) : hash_;
       this.setValue(this.self.update.verificationKey, { hash, data });
     }
     this.setValue(this.self.update.permissions, Permissions.default());
-    this.sign(zkappKey, true);
+    this.sign(zkappKey);
   }
 
-  sign(zkappKey?: PrivateKey, fallbackToZeroNonce?: boolean) {
-    this.self.signInPlace(zkappKey, fallbackToZeroNonce);
+  sign(zkappKey?: PrivateKey) {
+    this.self.sign(zkappKey);
   }
 
-  private executionState(): ExecutionState {
-    if (!Mina.currentTransaction.has()) {
+  private executionState(): Party {
```

Comment:
this was changed to use the party from `smartContractContext` if we're inside a smart contract, so it's less confusing where the `this.self` party originates (previously, it was the place where `this.self` was called first)

we also don't auto-attach the party to the transaction anymore in that case

Reply:
> Why is this.self not being added anymore?

I guess cause it's happening here?